src/main/java/io/temporal/samples/dsl/DynamicDslWorkflow.java

```diff
@@ -0,0 +1,203 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.dsl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.serverlessworkflow.api.actions.Action;
+import io.serverlessworkflow.api.events.OnEvents;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.states.EventState;
+import io.serverlessworkflow.api.states.OperationState;
+import io.serverlessworkflow.api.states.SwitchState;
+import io.serverlessworkflow.api.switchconditions.DataCondition;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.common.converter.EncodedValues;
+import io.temporal.workflow.*;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+
+public class DynamicDslWorkflow implements DynamicWorkflow {
+
+  private io.serverlessworkflow.api.Workflow dslWorkflow;
+  private JsonNode workflowData;
+  private Logger logger = Workflow.getLogger(DynamicDslWorkflow.class);
+  ActivityStub activities;
+
+  @Override
+  public Object execute(EncodedValues args) {
+    // Get first input and convert to SW Workflow object
+    String dslWorkflowId = args.get(0, String.class);
+    String dslWorkflowVersion = args.get(1, String.class);
+    // Get second input which is set to workflowData
+    workflowData = args.get(2, JsonNode.class);
+
+    dslWorkflow = DslWorkflowCache.getWorkflow(dslWorkflowId, dslWorkflowVersion);
```

Comment:
Add a comment that using a global shard workflow object is only allowed because it is read only. There is also an assumption that it is always exactly the same across workflow worker restarts.

Reply:
https://github.com/temporalio/samples-java/pull/153